packages/@aws-cdk/cdk/lib/app.ts

```diff
@@ -34,73 +33,49 @@ export class App extends Root {
   /**
    * Runs the program. Output is written to output directory as specified in the request.
    */
-  public run(): void {
+  public run(): ISynthesisSession {
+    // this app has already been executed, no-op for you
```

Comment:
What's the use case for the single-run behavior?

I don't think this is necessary, and it breaks use cases that loads of people are asking us about, where they run the CDK app and the toolkit in the same process. I mean, I know they're not officially supported right now, but I see no reason to make it completely impossible, just because we can, for a benefit that I can't really see.

Reply:
We can't ensure or enforce that mutations are idempotent, which implies that an app is "single use" and if someone calls `run()` again, they simply get the same result. This is also way I am "locking" the session after it's complete.